paws_collector.js

```diff
@@ -491,46 +492,44 @@ class PawsCollector extends AlAwsCollector {
             return callback(sendError, errorString);
         });
     }
-
+    /**
+     * Split data per 10K messages batch.
+     * Call the processLog to send Logmsgs to ingest
+     * @param {*} logs 
+     * @param {*} privCollectorState 
+     * @param {*} nextInvocationTimeout 
+     * @param {*} callback 
+     */
     batchLogProcess(logs, privCollectorState, nextInvocationTimeout, callback) {
         let collector = this;
-        return collector.processLog(logs, collector.pawsFormatLog.bind(collector), null,  async(err) => {
-           
-            if (err) {
-                /**
-                 * MS api return the logs but if build payload size > 10MB, and split logs array the in two part.
-                 * Call the processLog again to send Logmsgs to ingest. 
-                 */
-                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
-
-                    const half = Math.floor(logs.length / 2);
-                    const indexArray = [
-                        { start: 0, stop: half },
-                        { start: half, stop: logs.length }
-                    ];
-
-                    let promises = indexArray.map((logpart) => {
-                        return new Promise((resolve, reject) => {
-                            collector.batchLogProcess(logs.slice(logpart.start, logpart.stop), privCollectorState, nextInvocationTimeout, (err, res) => {
-                                if (err) {
-                                    reject(err);
-                                }
-                                else {
-                                    resolve(res);
-                                }
-                            });
-                        });
-                    });
-                    await Promise.all(promises);
-                    return callback(null, privCollectorState, nextInvocationTimeout);
-                } else {
-                    collector.reportErrorToIngestApi(err, () => {
-                        return callback(err);
-                    });
-                }
-            } else {
-                return callback(null, privCollectorState, nextInvocationTimeout);
+        let indexArray = [];
+        if (logs.length > MAX_LOG_BATCH_SIZE) {
```

Comment:
this `if..else` statement is unnecessary. if `slice.stop` is more than array length then the whole array is returned.